src/main/java/com/exadel/sober/controllers/PromiseController.java

```diff
@@ -0,0 +1,29 @@
+package com.exadel.sober.controllers;
+
+import com.exadel.sober.models.Promise;
+import com.exadel.sober.repositories.PromiseRepository;
+import com.exadel.sober.repositories.UserRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestController
+@CrossOrigin
+@RequestMapping(path="/promise")
+public class PromiseController {
+
+    @Autowired
+    private PromiseRepository promiseRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @PostMapping(path="/deletePromise")
+    public @ResponseBody List<Promise> deletePromiseByPromiseId (@RequestParam Integer promiseId, @RequestParam String userName) {
+
+        promiseRepository.deleteById(promiseId);
+
+        return userRepository.findByName(userName).getPromises();
```

Comment:
1) this logic must not be here. Please replace it into some service.
2) deletePromissById - must only delete but not delete and get.
3) findByName - not the best idea because userName is not unique
4) Don't forget about exceptions. How to know that promise has been deleted? For that you must use ResponceEntity with statusCode and in case all fine - you will get success statusCode in the client side - you will make another call and get all Promises. If something going wrong and you will not delete promise - send another ResponceEntity with responceCode and error message to the client